### DISCUSSION TITLE
How do I control the meta data in HTML <head>?

### DISCUSSION THREAD

**Discussion by jysung** [necolas/react-native-web#1974] [Q&A] [answered]
I am trying to generate a dynamic preview (on Twitter/FB/etc) per page depending on content.
It seems that these services expect these `<meta>` tag data to be there in `<head>` before javascript is run.

Because our `react-native-web` generates a single HTML file for every page, I cannot dynamically generate metadata for each page.

Is there any good way to solve this issue?

**Comment by necolas** [comment, marked as answer]
React Native for Web doesn't control the html shell. You can use any existing DOM solution for modifying head content, e.g., react-helmet
